test/extensions/filters/http/wasm/wasm_filter_stress_test.cc

```diff
@@ -0,0 +1,282 @@
+#include "envoy/server/wasm_config.h"
+
+#include "test/stress/stress_test.h"
+
+using namespace Envoy::Stress;
+
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+
+static const std::string STAT_PREFIX{"wasm_filter_stress_test"};
+
+class WasmStressTest : public Stress::StressTest,
+                       public testing::TestWithParam<
+                           std::tuple<std::string, std::string, std::string, std::string>> {
+public:
+  WasmStressTest()
+      : StressTest(::ipVersion(std::get<3>(GetParam())), ::httpType(std::get<2>(GetParam()))),
+        wasm_vm_{std::get<0>(GetParam())}, wasm_lang_{std::get<1>(GetParam())} {}
+
+  const std::string& wasmVM() const { return wasm_vm_; }
+
+  const std::string& wasmLang() const { return wasm_lang_; }
+
+private:
+  std::string wasm_vm_;
+  std::string wasm_lang_;
+};
+
+class GrpcWasmStressTest : public WasmStressTest {};
+
+class HttpWasmStressTest : public WasmStressTest {};
+
+INSTANTIATE_TEST_SUITE_P(RuntimesAndLanguages, GrpcWasmStressTest,
+                         testing::Combine(testing::Values(
+#if defined(ENVOY_WASM_V8) && defined(ENVOY_WASM_WAVM)
+                                              "envoy.wasm.vm.v8", "envoy.wasm.vm.wavm"
+#elif defined(ENVOY_WASM_V8)
+                                              "envoy.wasm.vm.v8"
+#elif defined(ENVOY_WASM_WAVM)
+                                              "envoy.wasm.vm.wavm"
+#endif
+                                              ),
+                                          testing::Values("cpp"), testing::Values("http2"),
+                                          testing::Values("IPv4", "IPv6")));
+
+INSTANTIATE_TEST_SUITE_P(RuntimesAndLanguages, HttpWasmStressTest,
+                         testing::Combine(testing::Values(
+#if defined(ENVOY_WASM_V8) && defined(ENVOY_WASM_WAVM)
+                                              "envoy.wasm.vm.v8", "envoy.wasm.vm.wavm"
+#elif defined(ENVOY_WASM_V8)
+                                              "envoy.wasm.vm.v8"
+#elif defined(ENVOY_WASM_WAVM)
+                                              "envoy.wasm.vm.wavm"
+#endif
+                                              ),
+                                          testing::Values("cpp"), testing::Values("http1", "http2"),
+                                          testing::Values("IPv4", "IPv6")));
+
+TEST_P(GrpcWasmStressTest, CalloutHappyPath) {
+  constexpr uint32_t connections_to_initiate = 30;
+  constexpr uint32_t requests_to_send = 30 * connections_to_initiate;
+  const std::string wasm_file = absl::StrCat(
+      TestEnvironment::runfilesDirectory(),
+      "/test/extensions/filters/http/wasm/test_data/grpc_callout_", wasmLang(), ".wasm");
+  // Must match cluster name in the wasm bundle:
+  const std::string callout_cluster_name{"callout_cluster"};
```

Comment:
@mandarjog Here is an example wasm test case.  The link between this test case and the wasm bundle is this 'callout_cluster' name.  

This test case creates two clusters:  a 'callout_cluster' to simulate some random upstream that the wasm bundle can callout to and a 'origin_cluster' which the data plane request will be forwarded to.    The grpc_callout_cpp.cc file passes the 'callout_cluster' name to the grpc library and Envoy does the rest